src/arrayMethodJoin.js

```diff
@@ -5,7 +5,19 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let joinedArray = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] !== null && this[i] !== undefined) {
+        joinedArray += this[i];
+      }
+
+      if (i < this.length - 1) {
+        joinedArray += separator !== undefined ? separator : ',';
```

Comment:
this is not the best way to use ternary operator. You can give default value to the parameter separator instead